File: ios/Orcha/Screens/GitHubIssueDetailScreen.swift
import SwiftUI

/// Issue detail — title + state, labels + assignees, the body (ChatMarkdownView),
/// the recent comment thread (oldest-first), and an open-on-GitHub link.
/// Start-from-detail lives in the toolbar. Same locally-owned phase + graceful-off
/// contract as the PR detail.
struct GitHubIssueDetailScreen: View {
    @Environment(AppModel.self) private var model
    @Environment(\.palette) private var p
    let number: Int

    @State private var phase: GitHubIssueDetailPhase = .loading
    @State private var showStartPicker = false
    @State private var startedTask: String?

    var body: some View {
        Group {
            switch phase {
            case .loading:
                loadingState
            case let .unavailable(reason, detail):
                GitHubDetailUnavailable(reason: reason, detail: detail) { await load() }
            case let .failed(message):
                GitHubDetailFailed(message: message) { await load() }
            case let .loaded(_, issue):
                loaded(issue)
            }
        }
        .navigationTitle("Issue #\(number)")
        .navigationBarTitleDisplayMode(.inline)
        .toolbar {
            if case .loaded = phase {
                ToolbarItem(placement: .topBarTrailing) {
                    Button("Start", systemImage: "play.fill") { showStartPicker = true }
                        .disabled(model.actionInFlight)
                }
            }
        }
        .sheet(isPresented: $showStartPicker) {
            if case let .loaded(_, issue) = phase {
                GitHubStartPickerSheet(
                    kind: .issues, number: issue.number, title: issue.title,
                    bodyExcerpt: String(issue.bodyMarkdown.prefix(200)), htmlUrl: issue.htmlUrl
                ) { response in
                    startedTask = response.taskId
                }
            }
        }
        .navigationDestination(item: $startedTask) { taskId in
            TaskDetailScreen(taskId: taskId)
        }
        .task { await load() }
    }

    private func load() async {
        phase = await model.loadGithubIssueDetail(number)
    }

    private var loadingState: some View {
        ScrollView {
            VStack(spacing: 12) {
                SkeletonBlock(height: 60)
                SkeletonBlock(height: 160)
            }
            .padding(16)
        }
    }

    private func loaded(_ issue: GitHubIssueDetail) -> some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 14) {
                header(issue)
                if !issue.bodyMarkdown.isEmpty {
                    section("Description") {
                        ChatMarkdownView(text: issue.bodyMarkdown)
                    }
                }
                commentsSection(issue)
                if let url = issue.htmlUrl.flatMap(URL.init(string:)) {
                    OpenOnGitHubLink(url: url)
                }
            }
            .padding(16)
        }
        .refreshable { await load() }
    }

    private func header(_ issue: GitHubIssueDetail) -> some View {
        VStack(alignment: .leading, spacing: 10) {
            Text(issue.title)
                .font(p.uiFont(19, .bold))
                .foregroundStyle(p.text)
                .fixedSize(horizontal: false, vertical: true)
            HStack(spacing: 6) {
                StatusPill(status: issue.state, domain: .task)
            }
            if !issue.labels.isEmpty {
                ScrollView(.horizontal, showsIndicators: false) {
                    HStack(spacing: 6) {
                        ForEach(issue.labels, id: \.self) { GitHubLabelChip(label: $0) }
                    }
                }
            }
            HStack(spacing: 6) {
                if let author = issue.authorLogin {
                    AgentAvatar(alias: author, human: true, githubLogin: author, size: 22)
                    Text(author).font(p.uiFont(12)).foregroundStyle(p.text2)
                }
                if !issue.assignees.isEmpty {
                    MetaTag(text: "assigned: \(issue.assignees.joined(separator: ", "))")
                }
                Spacer()
                Text(MobileUx.agoLabel(issue.updatedAt).map { "updated \($0)" } ?? "")
                    .font(p.uiFont(11)).foregroundStyle(p.faint)
            }
        }
    }

    @ViewBuilder
    private func commentsSection(_ issue: GitHubIssueDetail) -> some View {
        section("Comments · \(issue.commentsCount)") {
            if issue.comments.isEmpty {
                Text(issue.commentsCount > 0
                     ? "The comment thread couldn't be loaded."
                     : "No comments yet.")
                    .font(p.uiFont(13)).foregroundStyle(p.muted)
            } else {
                VStack(alignment: .leading, spacing: 12) {
                    if issue.commentsCount > issue.comments.count {
                        Text("Showing the most recent \(issue.comments.count) of \(issue.commentsCount) comments.")
                            .font(p.uiFont(11)).foregroundStyle(p.faint)
                    }
                    ForEach(issue.comments) { comment in
                        VStack(alignment: .leading, spacing: 4) {
                            HStack(spacing: 6) {
                                AgentAvatar(
                                    alias: comment.authorLogin ?? "?",
                                    human: true, githubLogin: comment.authorLogin, size: 22
                                )
                                Text(comment.authorLogin ?? "someone")
                                    .font(p.uiFont(13, .semibold))
                                    .foregroundStyle(p.text)
                                Spacer()
                                Text(MobileUx.agoLabel(comment.createdAt) ?? "")
                                    .font(.system(size: 10.5, design: .monospaced))
                                    .foregroundStyle(p.faint)
                            }
                            ChatMarkdownView(text: comment.bodyMarkdown)
                        }
                    }
                }
            }
        }
    }

    private func section(_ title: String, @ViewBuilder content: () -> some View) -> some View {
        VStack(alignment: .leading, spacing: 8) {
            SectionH(title: title)
            OrchaCard { content() }
        }
    }
}

File: android/app/src/main/java/io/openorcha/mobile/ui/screens/GitHubIssueDetailScreen.kt
package io.openorcha.mobile.ui.screens

/** Issue detail — Android parity of iOS `GitHubIssueDetailScreen.swift`. Title + state,
 *  labels + assignees, the raw markdown body, the recent comment thread (oldest-first),
 *  and an open-on-GitHub link. Start-from-detail lives in the top bar. */

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.material3.TopAppBar
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.material3.pulltorefresh.PullToRefreshBox
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import io.openorcha.mobile.data.GitHubComment
import io.openorcha.mobile.data.GitHubIssueDetail
import io.openorcha.mobile.domain.GitHubHubKind
import io.openorcha.mobile.domain.GitHubIssueDetailPhase
import io.openorcha.mobile.domain.MobileUx
import io.openorcha.mobile.ui.OrchaUiState
import io.openorcha.mobile.ui.components.Avatar
import io.openorcha.mobile.ui.components.AvatarSize
import io.openorcha.mobile.ui.components.GitHubLabelChip
import io.openorcha.mobile.ui.components.MetaTag
import io.openorcha.mobile.ui.components.OrchaCard
import io.openorcha.mobile.ui.components.SectionH
import io.openorcha.mobile.ui.components.Skeleton
import io.openorcha.mobile.ui.components.StatusDomain
import io.openorcha.mobile.ui.components.StatusPill
import io.openorcha.mobile.ui.components.MarkdownText
import io.openorcha.mobile.ui.icons.OrchaIcons
import io.openorcha.mobile.ui.theme.Orcha

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun GitHubIssueDetailScreen(
    state: OrchaUiState,
    onBack: () -> Unit,
    onRefresh: () -> Unit,
    onStart: (agentId: String?) -> Unit,
) {
    val p = Orcha.palette
    val number = state.githubIssueNumber ?: 0
    var showStartSheet by remember { mutableStateOf(false) }

    Scaffold(
        containerColor = Color.Transparent,
        topBar = {
            TopAppBar(
                colors = TopAppBarDefaults.topAppBarColors(containerColor = Color.Transparent),
                title = { Text("Issue #$number") },
                navigationIcon = { IconButton(onClick = onBack) { Icon(OrchaIcons.ArrowBack, "Back") } },
                actions = {
                    if (state.githubIssueDetailPhase is GitHubIssueDetailPhase.Loaded) {
                        TextButton(onClick = { showStartSheet = true }, enabled = !state.actionInFlight) { Text("Start") }
                    }
                },
            )
        },
    ) { padding ->
        Column(Modifier.fillMaxSize().padding(padding)) {
            when (val phase = state.githubIssueDetailPhase) {
                is GitHubIssueDetailPhase.Loading -> GitHubDetailLoading()
                is GitHubIssueDetailPhase.Unavailable -> PullToRefreshBox(isRefreshing = false, onRefresh = onRefresh) {
                    GitHubUnavailableState(phase.reason, phase.detail)
                }
                is GitHubIssueDetailPhase.Failed -> PullToRefreshBox(isRefreshing = false, onRefresh = onRefresh) {
                    GitHubFailedState(phase.message, onRefresh)
                }
                is GitHubIssueDetailPhase.Loaded -> PullToRefreshBox(isRefreshing = false, onRefresh = onRefresh) {
                    IssueDetailBody(phase.issue)
                }
            }
        }
    }

    if (showStartSheet) {
        GitHubStartSheet(
            kind = GitHubHubKind.Issues,
            number = number,
            agents = state.snapshot?.agents.orEmpty().filter { it.kind == "ai" && it.terminatedAt == null },
            busy = state.actionInFlight,
            onDismiss = { showStartSheet = false },
            onConfirm = { agentId -> showStartSheet = false; onStart(agentId) },
        )
    }
}

@Composable
private fun GitHubDetailLoading() {
    Column(Modifier.fillMaxSize().padding(16.dp), verticalArrangement = Arrangement.spacedBy(12.dp)) {
        Skeleton(height = 60.dp)
        Skeleton(height = 160.dp)
    }
}

@Composable
private fun IssueDetailBody(issue: GitHubIssueDetail) {
    val p = Orcha.palette
    LazyColumn(
        modifier = Modifier.fillMaxSize(),
        contentPadding = PaddingValues(16.dp),
        verticalArrangement = Arrangement.spacedBy(14.dp),
    ) {
        item { IssueHeader(issue) }
        if (issue.bodyMarkdown.isNotBlank()) {
            item {
                Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                    SectionH("Description")
                    OrchaCard { MarkdownText(issue.bodyMarkdown) }
                }
            }
        }
        item { SectionH("Comments", "${issue.commentsCount}") }
        if (issue.comments.isEmpty()) {
            item {
                OrchaCard {
                    Text(
                        if (issue.commentsCount > 0) "The comment thread couldn't be loaded." else "No comments yet.",
                        color = p.muted,
                    )
                }
            }
        } else {
            if (issue.commentsCount > issue.comments.size) {
                item {
                    Text(
                        "Showing the most recent ${issue.comments.size} of ${issue.commentsCount} comments.",
                        style = MaterialTheme.typography.labelMedium, color = p.faint,
                    )
                }
            }
            items(issue.comments) { comment -> CommentCard(comment) }
        }
        issue.htmlUrl?.let { url -> item { OpenOnGitHubLink(url) } }
    }
}

@Composable
private fun IssueHeader(issue: GitHubIssueDetail) {
    val p = Orcha.palette
    Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
        Text(issue.title, style = MaterialTheme.typography.titleLarge.copy(fontWeight = FontWeight.W700), color = p.text)
        StatusPill(issue.state, StatusDomain.Task)
        if (issue.labels.isNotEmpty()) {
            Row(horizontalArrangement = Arrangement.spacedBy(6.dp)) { issue.labels.forEach { GitHubLabelChip(it) } }
        }
        Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(6.dp)) {
            issue.authorLogin?.let { author ->
                Avatar(author, human = true, size = AvatarSize.Sm)
                Text(author, style = MaterialTheme.typography.bodyMedium, color = p.text2)
            }
            if (issue.assignees.isNotEmpty()) MetaTag("assigned: ${issue.assignees.joinToString(", ")}")
            Spacer(Modifier.weight(1f))
            Text(
                MobileUx.agoLabel(issue.updatedAt)?.let { "updated $it" } ?: "",
                style = MaterialTheme.typography.labelMedium, color = p.faint,
            )
        }
    }
}

@Composable
private fun CommentCard(comment: GitHubComment) {
    val p = Orcha.palette
    OrchaCard {
        Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(6.dp)) {
            Avatar(comment.authorLogin ?: "?", human = true, size = AvatarSize.Sm)
            Text(comment.authorLogin ?: "someone", style = MaterialTheme.typography.titleSmall, color = p.text)
            Spacer(Modifier.weight(1f))
            Text(MobileUx.agoLabel(comment.createdAt) ?: "", style = MaterialTheme.typography.labelSmall, color = p.faint)
        }
        MarkdownText(comment.bodyMarkdown)
    }
}
